pkg/console/controllers/oauthclients/oauthclients.go

```diff
@@ -0,0 +1,205 @@
+	secretsub "github.com/openshift/console-operator/pkg/console/subresource/secret"
+	"github.com/openshift/console-operator/pkg/crypto"
+)
+
+type oauthClientsController struct {
+	oauthClient    oauthv1client.OAuthClientsGetter
+	operatorClient v1helpers.OperatorClient
+	secretsClient  corev1client.SecretsGetter
+
+	oauthClientLister     oauthv1lister.OAuthClientLister
+	consoleOperatorLister operatorv1listers.ConsoleLister
+	routesLister          routev1listers.RouteLister
+	ingressConfigLister   configv1lister.IngressLister
+	targetNSSecretsLister corev1listers.SecretLister
+
+	statusHandler status.StatusHandler
+}
+
+func NewOAuthClientsController(
+	operatorClient v1helpers.OperatorClient,
+	oauthClient oauthv1client.OAuthClientsGetter,
+	secretsClient corev1client.SecretsGetter,
+	oauthClientInformer oauthv1informers.OAuthClientInformer,
+	consoleOperatorInformer operatorv1informers.ConsoleInformer,
+	routeInformer routev1informers.RouteInformer,
+	ingressConfigInformer configv1informers.IngressInformer,
+	targetNSsecretsInformer corev1informers.SecretInformer,
+	recorder events.Recorder,
+) factory.Controller {
+	c := oauthClientsController{
+		oauthClient:    oauthClient,
+		operatorClient: operatorClient,
+		secretsClient:  secretsClient,
+
+		oauthClientLister:     oauthClientInformer.Lister(),
+		consoleOperatorLister: consoleOperatorInformer.Lister(),
+		routesLister:          routeInformer.Lister(),
+		ingressConfigLister:   ingressConfigInformer.Lister(),
+		targetNSSecretsLister: targetNSsecretsInformer.Lister(),
+
+		statusHandler: status.NewStatusHandler(operatorClient),
+	}
+
+	return factory.New().
+		WithSync(c.sync).
+		WithInformers(
+			consoleOperatorInformer.Informer(),
+			routeInformer.Informer(),
+			ingressConfigInformer.Informer(),
+			targetNSsecretsInformer.Informer(),
+		).
+		WithFilteredEventsInformers(
+			factory.NamesFilter(api.OAuthClientName),
+			oauthClientInformer.Informer(),
+		).
+		WithSyncDegradedOnError(operatorClient).
+		ToController("OAuthClientsController", recorder.WithComponentSuffix("oauth-clients-controller"))
+}
+
+func (c *oauthClientsController) sync(ctx context.Context, controllerContext factory.SyncContext) error {
+	if shouldSync, err := c.handleStatus(ctx); err != nil {
+		return err
+	} else if !shouldSync {
+		return nil
+	}
+
+	operatorConfig, err := c.consoleOperatorLister.Get("cluster")
+	if err != nil {
+		return err
+	}
+
+	ingressConfig, err := c.ingressConfigLister.Get("cluster")
+	if err != nil {
+		return err
+	}
+
+	routeName := api.OpenShiftConsoleRouteName
+	routeConfig := routesub.NewRouteConfig(operatorConfig, ingressConfig, routeName)
+	if routeConfig.IsCustomHostnameSet() {
+		routeName = api.OpenshiftConsoleCustomRouteName
+	}
+
+	_, consoleURL, _, routeErr := operator.GetActiveRouteInfo(ctx, c.routesLister, routeName)
+	if routeErr != nil {
+		return routeErr
+	}
+
+	clientSecret, _, secErr := c.syncSecret(ctx, operatorConfig, controllerContext.Recorder())
```

Comment:
If we don't need the second value, could we just update the `c.syncSecret` to only return what we need?
```suggestion
	clientSecret, secErr := c.syncSecret(ctx, operatorConfig, controllerContext.Recorder())
```